### PR TITLE
FIX remove _joblib_parallel_args dependency & update base_estimator_ to estimator_

### LINKED ISSUE #5: BUG Fix joblib dependency in sklearn
@sampan501 Can you share the commented-out lines here?

### LINKED ISSUE #7: BUG Fix parameter name in forest initialization
@rflperry Observed in #6 

```python
TypeError: ForestClassifier.__init__() missing 1 required positional argument: 'estimator'
```

### PR DESCRIPTION
**Pull request by PSSF23**
Close #5 & Close #7 

- Remove `_joblib_parallel_args` from code
- Update `requirements.txt` with `joblib` version `1.0.0`
- Update `base_estimator_` to `estimator_`

### CODE REVIEW

**Review by PSSF23 — COMMENTED** [review]
Bug that happens with `pytest`:

```python
TypeError: ForestClassifier.__init__() missing 1 required positional argument: 'estimator'
```

Looks like another dependency issue?

File: honest_forests/estimators/forest.py [hunk 1]
```diff
@@ -446,7 +445,6 @@ def _predict_proba(self, X, indices=None, impute_missing=None):
         Parallel(
             n_jobs=n_jobs,
             verbose=self.verbose,
-            **_joblib_parallel_args(require="sharedmem")
```
Comment (sampan501) [inline honest_forests/estimators/forest.py:449]: I think this line should be `require="sharedmem"`

**Review by sampan501 — CHANGES_REQUESTED** [review]
just 1 change to maintain functionality

File: honest_forests/estimators/forest.py (hunk 1 above)
Comment (PSSF23) [reply honest_forests/estimators/forest.py:449]: sure. will change that.

**Review by PSSF23 — COMMENTED** [review]
Nice. Checks passed.

**Review by sampan501 — APPROVED** [review]

**Pull request merged**